src/main/java/game/Game.java

```diff
@@ -66,6 +67,19 @@ public void updatePlayerState() {
         this.player.decrementIframes();
     }
 
+    public void teleportPlayer() {
```

Comment:
I think this method violates clean architecture. You are bringing entities up into the controller (game). 

Rather than doing this, you should perhaps move player position into PlayerState, and have it be modified inside the objectStateManager.

Reply:
To clarify. Interactable interface's whole purpose is to modify the playerstate. If you were to move player position into player state (this will require a lot of code to be updated, but the refactoring should be trivial), then you can simply change the player position there in your teleporter thing.